### PR TITLE
fix(ua-restriction): refine plugin configuration check logic

### PR DESCRIPTION
**Pull request by zuiyangqingzhou**
fix: [5724](https://github.com/apache/apisix/issues/5724)

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/ua-restriction.lua [hunk 1]
```diff
@@ -34,11 +35,11 @@ local schema = {
             type = "boolean",
             default = false,
         },
-        allowlist = {
+        allow_list = {
```
Comment (tokers) [inline apisix/plugins/ua-restriction.lua:38]: It's a broken change.

**Comment by juzhiyuan** [comment]
@zuiyangqingzhou Hi, please also update this PR's description 😄

File: apisix/plugins/ua-restriction.lua [hunk 2]
```diff
@@ -34,11 +35,11 @@ local schema = {
...
             type = "array",
             minItems = 1
         },
-        denylist = {
+        deny_list = {
```
[the remark this answers is not in the document]
Comment (zuiyangqingzhou) [reply apisix/plugins/ua-restriction.lua:42]: Oh, I just don't think it's in line with the variable naming specification, and if the scope of the impact is too great, I'll change it back  @juzhiyuan

[the remark this answers is not in the document]
Comment (juzhiyuan) [reply apisix/plugins/ua-restriction.lua]: sure, we have a code style file 😄 https://github.com/apache/apisix/blob/master/CODE_STYLE.md#variable

**Review by spacewander — COMMENTED** [review]
Could you add a test in https://github.com/apache/apisix/blob/master/t/plugin/ua-restriction.t?

**Comment by zuiyangqingzhou** [comment]
> Could you add a test in https://github.com/apache/apisix/blob/master/t/plugin/ua-restriction.t?

OK

File: t/plugin/ua-restriction.t [hunk 3]
```diff
@@ -739,3 +739,51 @@ hello world
...
+            local ok, err = plugin.check_schema(conf)
+            if not ok then
+                ngx.say(err)
+            end
+
+            ngx.say("done")
+        }
+    }
+--- response_body
+property "allowlist" validation failed: expect array to have at least 1 items
```
Comment (spacewander) [inline t/plugin/ua-restriction.t:764]: Look like the conf is not rejected because of the bad regex? Maybe we need a wrong regex to test it.

Comment (zuiyangqingzhou) [reply t/plugin/ua-restriction.t]: No problem

**Comment by spacewander** [comment]
@zuiyangqingzhou 
Would you update the test to make CI pass?

**Comment by zuiyangqingzhou** [comment]
> @zuiyangqingzhou Would you update the test to make CI pass?
OK

**Comment by zuiyangqingzhou** [comment]
image
This should be caused by something else

File: t/plugin/ua-restriction.t [hunk 4]
```diff
@@ -783,5 +790,7 @@ load_full_data(): failed to check item data of [/apisix/routes] err:failed to ch
             ngx.say("done")
         }
     }
---- error_log
-load_full_data(): failed to check item data of [/apisix/routes] err:failed to check the configuration of plugin ua-restriction
+--- response_body
+property "denylist" validation failed: wrong type: expected array, got table
+done
+--- ONLY
```
Comment (membphis) [inline t/plugin/ua-restriction.t]: need to remove this line

**Review by tokers — APPROVED** [review]

File: t/plugin/ua-restriction.t (hunk 4 above)
Comment (zuiyangqingzhou) [reply t/plugin/ua-restriction.t]: yes, it has been deleted

**Review by membphis — APPROVED** [review]

**Comment by membphis** [comment]
I rerun the CI right now. one of the ci failed.

**Review by spacewander — APPROVED** [review]

**Pull request merged**